bubblewrap.c

```diff
@@ -1624,9 +1639,30 @@ main (int    argc,
       exit (0); /* Should not be reached, but better safe... */
     }
 
+  /* Child, in sandbox, privileged in the parent *or* in the user namespace.
+   *
+   * NOTE: This is always ptrace:able in the case of user namespaces
+   * (due to all parent namespaces having CAP_SYS_PTRACE in the child
+   * namespaces), but it should be ok as it has no permissions in the
+   * parent namespace.
```

Comment:
Clearly if bwrap is non-privileged then nothing is privileged. The interesting cases are 1) setuid bwrap + no user space, and 2) setuid + --unshare-user. In case 1, the child will have CAP_SYS_ADMIN in the host user namespace. In case 2, the child will have no caps in the host user namespace and *all* caps in the child user namespace.

In case 1 we rely on not being able to ptrace the child, but in case 2 that doesn't matter.